sklearn/metrics/tests/test_ranking.py

```diff
@@ -1650,6 +1650,26 @@ def test_top_k_accuracy_score_binary(y_score, k, true_score):
     assert score == score_acc == pytest.approx(true_score)
 
 
+@pytest.mark.parametrize('y_true, true_score, labels', [
+    (np.array([0, 1, 2, 3]), 0.5, None),
+    (np.array([0, 1, 1, 2]), 0.75, [0, 1, 2, 3]),
```

Comment:
For labels as an ndarray and labels as strings. Also, we do not need to test the `labels=None` case since it is already tested in elsewhere. 

```py
@pytest.mark.parametrize('y_true, true_score, labels', [
    (np.array(['a', 'e', 'e', 'a']), 0.75, ['a', 'b', 'd', 'e']),
    (np.array([0, 1, 1, 1]), 0.5, [0, 1, 2, 3]),
    (np.array([1, 3, 1, 3]), 0.25, [0, 1, 2, 3]),
])
@pytest.mark.parametrize("labels_as_ndarray", [True, False])
def test_top_k_accuracy_score_multiclass_with_labels(
        y_true, true_score, labels, labels_as_ndarray
):
	"""Test when y_true looks binary but the labels and y_score is multiclass."""
    if labels_as_ndarray:
        labels = np.asarray(labels)
	...
```

Reply:
Good point, I missed those. 

I added these 3 commits ([1](https://github.com/scikit-learn/scikit-learn/pull/19300/commits/e9b3d18075a1a06cad2789f592ee26839a1a0564), [2](https://github.com/scikit-learn/scikit-learn/pull/19300/commits/76681aa08ed1515327e25f8ac26ee801fcbd2341), [3](https://github.com/scikit-learn/scikit-learn/pull/19300/commits/7ea6aa2750f69e4b013cba4e3b3459dc45ca2bbe)) incorporating your suggestions.

Reply:
And the comment [in this commit](https://github.com/scikit-learn/scikit-learn/pull/19300/commits/1ff696c27470f8ee441e146cd1ac776385dd58fe).